Tasks/TasksListScreen.swift

```diff
@@ -5,116 +5,194 @@
+        // would be to run a function on a regular time interval that evicts all documents where 
+        // isSafeForEviction == true, not recommended to exceed once per day.
+        // See docs for best practices (link)
+        do {
+            try await dittoStore.execute(
+                query: "EVICT FROM tasks WHERE _id == :_id",
+                arguments: ["_id": taskToEvict._id]
+            )
+        } catch {
+            print("TaskListScreenVM.\(#function) - ERROR evicting task: \(error.localizedDescription)")
         }
-            
-        self.subscription = DittoManager.shared.ditto.store["tasks"]
-            .find(query).subscribe()
-        
-        self.liveQuery = DittoManager.shared.ditto.store["tasks"]
-            .find(query)
-            .observeLocal(eventHandler: {  docs, event in
-                print(event.description)
-                self.tasks = docs.map({ Task(document: $0) })
-                print(self.tasks)
-            })
-        DittoManager.shared.ditto.store["tasks"].find("isDeleted == true").evict()
+        taskToEdit = nil
     }
     
     public static func randomFakeFirstName() -> String {
         return TasksApp.firstNameList.randomElement()!
     }
     
-    func toggle(task: Task) {
-        DittoManager.shared.ditto.store["tasks"].findByID(task._id)
-            .update { mutableDoc in
-                guard let mutableDoc = mutableDoc else { return }
-                mutableDoc["isCompleted"].set(!mutableDoc["isCompleted"].boolValue)
+    func toggleComplete(task: TaskModel) {        
+        Task {
+            let isComplete = !task.isCompleted
+            let query = "UPDATE COLLECTION tasks (invitationIds MAP)"
+            + "SET isCompleted = :isCompleted WHERE _id == :_id"
+
+            do {
+                try await dittoStore.execute(
+                    query: query,
+                    arguments: ["isCompleted": isComplete, "_id": task._id]
+                )
+            } catch {
+                print("TaskListScreenVM.\(#function) - ERROR toggling task: \(error.localizedDescription)")
             }
+        }
     }
-    
-    func clickedInvite(task: Task) {
-        DittoManager.shared.ditto.store["tasks"].findByID(task._id)
-            .update { mutableDoc in
-                guard let mutableDoc = mutableDoc else { return }
-                var userId = TasksListScreenViewModel.randomFakeFirstName()
-                mutableDoc["invitationIds"][userId] = true
+
+    func clickedInvite(task: TaskModel)  {
+        let invitedUser = TasksListScreenViewModel.randomFakeFirstName()
+        let query = "UPDATE COLLECTION tasks (invitationIds MAP)"
+        + " SET invitationIds -> ( \(invitedUser) = true )"
+        + " WHERE _id == :_id"
+        
+        Task {
+            do {
+                try await dittoStore.execute(
+                    query: query,
+                    arguments: ["invitedUser": invitedUser, "_id": task._id]
```

Comment:
Looks like we can remove `"invitedUser": invitedUser` from the arguments.

```suggestion
                    arguments: ["_id": task._id]
```